api/applications.js

```diff
@@ -83,5 +84,152 @@ module.exports = [
                 }
             }
         }  
+    },
+    {
+        method: 'POST',
+        path: '/applications/{applicationId}/userSignup',
+        handler: async (request, h) => {
+            const uow = await request.app.getNewUoW();
+            const logger = request.server.app.logger;
+            const payload = request.payload;
+            let errors = [];
+
+            //send a confirmation email by default
+            let sendConfirmationEmail = true;
+            if (payload.sendConfirmationEmail != null) {
+                sendConfirmationEmail = payload.sendConfirmationEmail;
+            }
+
+            try {
+
+                logger.debug(`Creating user from survey`);
+                const payload = request.payload;
+
+                //set up models for each separate object
+                const userModel = {
+                    primaryEmailAddress: payload.primaryEmailAddress,
+                    firstName: payload.firstName,
+                    lastName: payload.lastName
+                };
+
+                let phones = payload.phones;
+                logger.debug(phones);
+                let organizations = payload.organizations;
+                logger.debug(organizations);
+
+                //begin the transaction
+                await uow.beginTransaction();
+
+                //if a user already has the submitted email, add it to errors and don't create it
+                const existingUser = await uow.usersRepository.getUserByEmail(userModel.primaryEmailAddress);
+                if (existingUser != null) {
+                    let err = `A user with email ${existingUser.primaryEmailAddress} already exists and was not created`
+                    errors.push(err);
+                    logger.debug(err);
+                }
+
+                //if an organization matching the new organization information, add it to errors and don't create it
+                let dbOrganizationIds = [];
```

Comment:
This stuff should all take place inside an `else` block for the above `if` statement on line 125. If a user already exists, don't even spend the time in this request working with the organizations.